environment.yml

```diff
@@ -8,16 +8,15 @@ channels:
   - conda-forge
 dependencies:
   - python=3.9 #need to pin to avoid issues with builds
-  - cython>=0.29.30
+  - cython>=3.0
   - compilers
   - openblas
   - nomkl
   - setuptools=59.2.0
-  - meson >= 0.64.0
   - ninja
   - pkg-config
-  - meson-python
-  - pip   # so you can use pip to install spin
+  - meson-python>=0.10.0
```

Comment:
not sure where this one came from, we already require >=0.13.1 in `pyproject.toml` - better to leave unpinned, there no use for a lower bound here

Let's leave `pip` I'd say, it's needed a lot for development, not only for installing `spin` but for, e.g., editable installs.